### PR TITLE
Add conformance tests to CAPV

### PR DESCRIPTION
**Pull request by aartij17**
**What this PR does / why we need it**:
This PR adds conformance tests to CAPV. 

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

_Please confirm that if this PR changes any image versions, then that's the sole change this PR makes._

**Release note**:

```release-note
NONE
```

### CODE REVIEW

File: test/e2e/common.go [hunk 1]
```diff
@@ -17,11 +17,55 @@ limitations under the License.
...
+	bootstrapClusterProxy framework.ClusterProxy
+
+	namespaces map[*corev1.Namespace]context.CancelFunc
+)
+
 func Byf(format string, a ...interface{}) {
 	By(fmt.Sprintf(format, a...))
 }
+
+func setupSpecNamespace(ctx context.Context, specName string, clusterProxy framework.ClusterProxy, artifactFolder string) (*corev1.Namespace, context.CancelFunc) {
```
Comment (srm09) [inline test/e2e/common.go:61]: Not necessary to bring this change in, use the CAPI E2E code as a library. Similar to the comment I have added below.

File: test/e2e/config/vsphere-ci.yaml [hunk 2]
```diff
@@ -169,6 +169,8 @@ variables:
   INIT_WITH_BINARY_V1ALPHA4: "https://github.com/kubernetes-sigs/cluster-api/releases/download/v0.4.4/clusterctl-{OS}-{ARCH}"
   INIT_WITH_KUBERNETES_VERSION: "v1.19.1"
   VSPHERE_INSECURE_CSI: "true"
+  KUBETEST_CONFIGURATION: "./data/kubetest/conformance.yaml"
+
```
Comment (srm09) [inline test/e2e/config/vsphere-ci.yaml:173]: nit: remove extra line

**Review by srm09 — CHANGES_REQUESTED** [review]
Added inline comments to use CAPI e2e code as a library

File: test/e2e/common.go [hunk 3]
```diff
@@ -17,9 +17,38 @@ limitations under the License.
 package e2e
 
 import (
+	"context"
 	"fmt"
 
 	. "github.com/onsi/ginkgo" // nolint:gci
+	corev1 "k8s.io/api/core/v1"
```
Comment (srm09) [inline test/e2e/common.go:24]: In general a comment about this file, any reason why we are moving the test vars to the `common.go` file. IMO the purpose of the file is to host constants and functions to be shared between different tests.

**Comment by aartij17** [comment]
@srm09 I had moved the constants to `common.go` cause I was reusing them in another file(`k8s_conformance.go`). Exported variables/functions declared in a `*_test.go` couldn't be used in another file. 
But now I removed `k8s_conformance.go`. I will move the variables back to the original file.

**Review by srm09 — COMMENTED** [review]
Looks good to me except for a nit. Thanks for the great work.

File: test/e2e/k8s_conformance_test.go [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+
+var _ = Describe("When testing K8S conformance [Conformance]", func() {
+	capi_e2e.K8SConformanceSpec(ctx, func() capi_e2e.K8SConformanceSpecInput {
+		return capi_e2e.K8SConformanceSpecInput{
+			E2EConfig:             e2eConfig,
+			ClusterctlConfigPath:  clusterctlConfigPath,
+			BootstrapClusterProxy: bootstrapClusterProxy,
+			ArtifactFolder:        artifactFolder,
+			SkipCleanup:           skipCleanup,
+		}
```
Comment (srm09) [inline test/e2e/k8s_conformance_test.go:32]: Whoops, include the new flavor that was created for this particular test which provisions beefy k8s worker nodes to run the tests without a hitch.

File: test/e2e/k8s_conformance_test.go [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+)
+
+var _ = Describe("When testing K8S conformance [Conformance]", func() {
+	capi_e2e.K8SConformanceSpec(ctx, func() capi_e2e.K8SConformanceSpecInput {
+		return capi_e2e.K8SConformanceSpecInput{
+			E2EConfig:             e2eConfig,
+			ClusterctlConfigPath:  clusterctlConfigPath,
+			BootstrapClusterProxy: bootstrapClusterProxy,
+			ArtifactFolder:        artifactFolder,
+			SkipCleanup:           skipCleanup,
```
Comment (srm09) [inline test/e2e/k8s_conformance_test.go:31]:
```suggestion
			SkipCleanup:           skipCleanup,
			Flavor:                       "conformance",
```

File: test/e2e/data/infrastructure-vsphere/kustomization/conformance/worker-node-size.yaml [hunk 6]
```diff
@@ -0,0 +1,25 @@
+---
+apiVersion: infrastructure.cluster.x-k8s.io/v1beta1
+kind: VSphereMachineTemplate
+metadata:
+  name: ${CLUSTER_NAME}-worker
+  namespace: ${NAMESPACE}
+spec:
+  template:
+    spec:
+      cloneMode: linkedClone
```
Comment (geetikabatra) [inline test/e2e/data/infrastructure-vsphere/kustomization/conformance/worker-node-size.yaml:10]: Is this specific to Vsphere? @srm09 and @aartij17 ?

Comment (srm09) [reply test/e2e/data/infrastructure-vsphere/kustomization/conformance/worker-node-size.yaml:10]: The template is specific to vSphere. Does that answer your question?

**Pull request merged**